src/sync_protocol/include/persistent_queue_storage.hpp

```diff
@@ -12,6 +12,31 @@
 #include "ipersistent_queue_storage.hpp"
 #include "sqlite3Wrapper.hpp"
 
+/// @brief Defines the synchronization status of a persisted message.
+enum class SyncStatus : int
+{
+    PENDING = 0,        ///< The message is waiting to be synchronized.
+    SYNCING = 1,        ///< The message is currently being synchronized.
+    SYNCING_UPDATED = 2 ///< The message is being synchronized and its contents have been updated.
+};
+
+/// @brief Tracks the creation state of a persisted message, particularly for newly created items.
+enum class CreateStatus : int
+{
+    NO_NEW_CREATED = 0,         ///< The message existed prior to the current session; it was not newly created.
+    NEW_CREATED = 1,            ///< The message was newly created during the current session.
+    DELETED_AFTER_CREATED = 2   ///< The message was newly created, but then deleted before it could be synchronized.
```

Comment:
```suggestion
    EXISTING = 0,     ///< The message existed prior to the current session; it was not newly created.
    NEW = 1,          ///< The message was newly created during the current session.
    NEW_DELETED = 2   ///< The message was newly created, but then deleted before it could be synchronized.
```
I think this is easier to understand.

Reply:
Done!